### PR TITLE
Add additional llm request validation

### PR DESCRIPTION
**Pull request by michalkulakowski**
### 🛠 Summary

JIRA/Issue if applicable.
Describe the changes.

### 🧪 Checklist

- [ ] Unit tests added.
- [ ] The documentation updated.
- [ ] Change follows security best practices.
``

### CODE REVIEW

**Review by dkalinowski — APPROVED** [review]

**Review by dtrawins — APPROVED** [review]

File: src/llm/http_llm_calculator.cc [hunk 1]
```diff
@@ -160,16 +162,30 @@ class OpenAIChatCompletionsRequest {
...
         // max_tokens: int; optional
         it = this->doc.FindMember("max_tokens");
         if (it != this->doc.MemberEnd()) {
-            if (!it->value.IsInt())
-                return absl::InvalidArgumentError("max_tokens is not an integer");
-            this->maxTokens = it->value.GetInt();
-            if (this->maxTokens.value() <= 0)
-                return absl::InvalidArgumentError("max_tokens value should be greater than 0");
+            if (!it->value.IsUint())
+                return absl::InvalidArgumentError("max_tokens is not an unsigned integer");
```
Comment (mzegla) [inline src/llm/http_llm_calculator.cc:178]: Does it work with 0? Could you check how OVMS will behave then?

File: src/llm/http_llm_calculator.cc [hunk 2]
```diff
@@ -46,6 +46,8 @@ namespace mediapipe {
 using chat_entry_t = std::unordered_map<std::string, std::string>;
 using chat_t = std::vector<chat_entry_t>;
 
+#define IGNORE_EOS_MAX_TOKENS_LIMIT 4000
```
Comment (mzegla) [inline src/llm/http_llm_calculator.cc:49]: Is it sufficient? Llama3 has 8k context size for example

File: src/llm/http_llm_calculator.cc [hunk 3]
```diff
@@ -253,27 +269,27 @@ class OpenAIChatCompletionsRequest {
...
         }
 
         // best_of: int; optional - defaults to 1
         // Extension, unsupported by OpenAI API, however supported by vLLM, supported in CB lib by mapping to group_size param
         it = this->doc.FindMember("best_of");
         if (it != this->doc.MemberEnd()) {
-            if (!it->value.IsInt())
-                return absl::InvalidArgumentError("best_of is not an integer");
-            this->bestOf = it->value.GetInt();
+            if (!it->value.IsUint())
```
Comment (mzegla) [inline src/llm/http_llm_calculator.cc:281]: What if 0 is passed?

File: src/llm/http_llm_calculator.cc [hunk 4]
```diff
@@ -253,27 +269,27 @@ class OpenAIChatCompletionsRequest {
...
         }
 
         // n: int; optional - defaults to 1
         // Supported by OpenAI API and vLLM, supported in CB lib by mapping to num_return_sequences param
         it = this->doc.FindMember("n");
         if (it != this->doc.MemberEnd()) {
-            if (!it->value.IsInt())
-                return absl::InvalidArgumentError("n is not an integer");
-            this->numReturnSequences = it->value.GetInt();
+            if (!it->value.IsUint())
```
Comment (mzegla) [inline src/llm/http_llm_calculator.cc:290]: What if there's 0?

File: src/llm/http_llm_calculator.cc (hunk 2 above)
Comment (dtrawins) [reply src/llm/http_llm_calculator.cc:49]: @mzegla  this is only for scenarios with ignore_eos so for testing purposes.

**Review by mzegla — APPROVED** [review]

**Pull request merged**